pkg/gatewayserver/grpc_nsgs.go

```diff
@@ -72,17 +72,22 @@ func (gs *GatewayServer) ScheduleDownlink(ctx context.Context, down *ttnpb.Downl
 			pathErrs = append(pathErrs, errNotConnected.WithAttributes("gateway_uid", uid))
 			continue
 		}
+		registerScheduleDownlinkAttempt(ctx, conn.Gateway(), down, conn.Frontend().Protocol())
+
 		connDown := deepcopy.Copy(down).(*ttnpb.DownlinkMessage) // Let the connection own the DownlinkMessage.
 		connDown.GetRequest().DownlinkPaths = nil                // And do not leak the downlink paths to the gateway.
 		rx1, rx2, delay, err := conn.ScheduleDown(path, connDown)
 		if err != nil {
 			logger.WithField("gateway_uid", uid).WithError(err).Debug("Failed to schedule on path")
 			pathErrs = append(pathErrs, errSchedulePath.WithCause(err).WithAttributes("gateway_uid", uid))
+			registerScheduleDownlinkFail(ctx, conn.Gateway(), err, conn.Frontend().Protocol())
 			continue
 		}
+
 		ctx = events.ContextWithCorrelationID(ctx, events.CorrelationIDsFromContext(conn.Context())...)
 		down.CorrelationIds = append(down.CorrelationIds, events.CorrelationIDsFromContext(ctx)...)
 		registerSendDownlink(ctx, conn.Gateway(), down, conn.Frontend().Protocol())
```

Comment:
`connDown` instead of `conn`, right ?

Reply:
Yeah the actual code is in a local stash, thanks